packages/evm/src/eof/container.ts

```diff
@@ -0,0 +1,367 @@
+import {
+  CODE_MIN,
+  CODE_SIZE_MIN,
+  CONTAINER_MAX,
+  CONTAINER_MIN,
+  CONTAINER_SIZE_MIN,
+  FORMAT,
+  INPUTS_MAX,
+  KIND_CODE,
+  KIND_CONTAINER,
+  KIND_DATA,
+  KIND_TYPE,
+  MAGIC,
+  MAX_HEADER_SIZE,
+  MAX_STACK_HEIGHT,
+  OUTPUTS_MAX,
+  TERMINATOR,
+  TYPE_DIVISOR,
+  TYPE_MAX,
+  TYPE_MIN,
+  VERSION,
+} from './constants.js'
+import { EOFError, validationError } from './errors.js'
+import { ContainerSectionType, verifyCode } from './verify.js'
+
+import type { EVM } from '../evm.js'
+
+export enum EOFContainerMode {
+  Default, // Default container validation
+  Initmode, // Initmode container validation (for subcontainers pointed to by EOFCreate)
+  TxInitmode, // Tx initmode container validation (for txs deploying EOF contracts)
+}
```

Comment:
Not fully sure if this is only about validation or not.

If it is I would suggest the following renaming for better clarity:

```ts
export enum EOFContainerValidationMode {
  Default, // Default container validation
  EOFCreateSubcontainerInit, // Initmode container validation (for subcontainers pointed to by EOFCreate)
  TxEOFContractDeployment, // Tx initmode container validation (for txs deploying EOF contracts)
}
```

(or similar, maybe take the bits out you like or evolve 🙂)

The current version is some mix out of being an enum for a "container type" (in the wider sense) and a "validation mode", which is a bit ambiguous.

Reply:
Yes those flags are confusing, I wanted to look if I can compress these in a single flag instead of 2 flags with 3 options (this leads to 9 unique combinations).

Reply:
I have to revisit renaming those validation modes. For the other flag, I have added a long comment explaining what it does.